example/lib/ui/device_detail_screen.dart

```diff
@@ -0,0 +1,159 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_reactive_ble/flutter_reactive_ble.dart';
+
+import '../src/utils.dart';
+import '../src/widgets.dart';
+
+final _serviceUuid = Uuid.parse("932C32BD-0000-47A2-835A-A8D455B859DD");
+final _charUuid = Uuid.parse("932C32BD-0002-47A2-835A-A8D455B859DD");
+
+class DeviceDetailScreen extends StatefulWidget {
+  final DiscoveredDevice device;
+
+  const DeviceDetailScreen({@required this.device}) : assert(device != null);
+
+  @override
+  State<StatefulWidget> createState() => _DetailState(device.id, device.name);
+}
+
+class _DetailState extends State<DeviceDetailScreen> {
+  final _ble = FlutterReactiveBle();
+
+  _DetailState(this.deviceId, this.deviceName);
+
+  final String deviceId;
+  final String deviceName;
+
+  Stream<ConnectionStateUpdate> _connectedDeviceStream;
+  Stream<List<int>> _charValueStream;
+  StreamSubscription _currentValueUpdateSubscription;
+
+  DeviceConnectionState _connectionState;
+  int _currentValue;
+
+  @override
+  void initState() {
+    super.initState();
+    _connectToDevice();
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _currentValueUpdateSubscription?.cancel();
+  }
+
+  void _connectToDevice() {
+    log("Connecting to $deviceName ($deviceId)");
+    _connectedDeviceStream = _ble
+        .connectToDevice(
+          id: deviceId,
+          connectionTimeout: const Duration(seconds: 5),
+        )
+        .asBroadcastStream();
+
+    _charValueStream = _connectedDeviceStream.where((device) => device.connectionState == DeviceConnectionState.connected).take(1).asyncExpand((d) {
+      log("Connected to $deviceName ($deviceId), getting characteristic");
+
+      return _ble.subscribeToCharacteristic(QualifiedCharacteristic(
+        characteristicId: _charUuid,
+        serviceId: _serviceUuid,
+        deviceId: deviceId,
+      ));
+    });
+
+    _currentValueUpdateSubscription = _charValueStream.listen((data) => _currentValue = data.first);
+
+    _connectedDeviceStream
+        .where((device) => device.connectionState == DeviceConnectionState.connected)
+        .first
+        .then((_) => Future<void>.delayed(const Duration(milliseconds: 20)))
```

Comment:
For Android maybe we should increase the delay to 50 or 100ms personally think 20 is a bit to short